File: java/decorator/StarbuzzCoffee/decorator/condiments/Soy.java
package decorator.condiments;

import component.Beverage;
import decorator.CondimentDecorator;

public class Soy extends CondimentDecorator {

    private Beverage beverage;

    public Soy(Beverage beverage) { this.beverage = beverage; }

    @Override
    public double cost() { return beverage.cost() + 0.15; }

    @Override
    public String getDescription() { return beverage.getDescription() + ", Soy"; }
}

File: kotlin/decorator/StarbuzzCoffee/decorator/condiments/Soy.kt
package decorator.condiments

import component.Beverage
import decorator.CondimentDecorator

class Soy(beverage: Beverage) : CondimentDecorator() {

    private var beverage: Beverage? = null

    init { this.beverage = beverage }

    override fun getDescription(): String { return beverage!!.getDescription() + ", Soy" }

    override fun cost(): Double { return beverage!!.cost() + 0.10 }

}